packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -227,6 +227,7 @@ export class OdspDocumentService implements IDocumentService {
 				}
 			},
 			(ops: ISequencedDocumentMessage[]) => this.opsReceived(ops),
+			this.storageManager?.isFirstSnapshotFromNetwork,
```

Comment:
I think it's fine today, but it's worth pointing out that this code depends a lot on sequencing of events. I.e. if somehow Loader layer would start calling connectToDeltaStorage() before (or in parallel) to loading snapshot (which feels like reasonable thing to do), then we would provide a wrong data here.
That said, same is true for very first line in this function (this.storageManager?.ops).

Should this code assert that this.storageManager ?.isFirstSnapshotFromNetwork !== undefined?
I.e. that we already have storageManager, and we also have snapshot when this call is made?
It's a weak way to enforce contract, but it's at least something (that I hope we will notice as broken if we change the logic).
Alternatively, we can add an error telemetry, though I'd personally prefer assert (more likely to be noticed as it will break UT, and smaller code size)